### PR TITLE
Use `upsert_all` and `insert_all` to reduce back-and-forth in costly migrations

### PR DESCRIPTION
**Pull request by ClearlyClaire**
Trying on 20k generated local users, the migrations went from about 23 seconds to about 2 seconds.

### CODE REVIEW

File: db/migrate/20240304090449_migrate_interaction_settings_to_policy.rb [hunk 1]
```diff
@@ -4,46 +4,50 @@ class MigrateInteractionSettingsToPolicy < ActiveRecord::Migration[7.1]
...
 
   # Dummy classes, to make migration possible across version changes
-  class Account < ApplicationRecord
-    has_one :user, inverse_of: :account
-    has_one :notification_policy, inverse_of: :account
-  end
-
   class User < ApplicationRecord
-    belongs_to :account
+    belongs_to :notification_policy, foreign_key: 'account_id', optional: true, inverse_of: false
```
Comment (mjankowski) [inline db/migrate/20240304090449_migrate_interaction_settings_to_policy.rb:8]: Does this also need a `primary_key` on it?

As written, I would think this is going to try to find `NotificationPolicy` records where the `users.account_id` value equals the `notification_policies.id` value ... maybe I'm missing something obvious here.

File: db/migrate/20240304090449_migrate_interaction_settings_to_policy.rb [hunk 2]
```diff
@@ -4,46 +4,50 @@ class MigrateInteractionSettingsToPolicy < ActiveRecord::Migration[7.1]
...
 
-  class NotificationPolicy < ApplicationRecord
-    belongs_to :account
-  end
+  class NotificationPolicy < ApplicationRecord; end
 
   def up
-    User.includes(account: :notification_policy).find_each do |user|
-      deserialized_settings = Oj.load(user.attributes_before_type_cast['settings'])
+    User.includes(:notification_policy).in_batches do |users|
```
Comment (mjankowski) [inline db/migrate/20240304090449_migrate_interaction_settings_to_policy.rb:14]: If I recall from the original migration - the audience of records we want here is like "Users who have selected one of the settings UI prefs before the notification policy migrations have run", right?

If so, could we even more narrowly select only the relevant users here, or is that impossible (or too complex or edge-casey) given the settings serialization.

File: db/migrate/20240304090449_migrate_interaction_settings_to_policy.rb (hunk 1 above)
Comment (ClearlyClaire) [reply db/migrate/20240304090449_migrate_interaction_settings_to_policy.rb:8]: Good catch! You're right, this needs `primary_key: 'account_id'`, fixing that.

File: db/migrate/20240304090449_migrate_interaction_settings_to_policy.rb (hunk 2 above)
Comment (ClearlyClaire) [reply db/migrate/20240304090449_migrate_interaction_settings_to_policy.rb:14]: Settings are serialized as a JSON blob in a `text` column, so I guess we could cast and do stuff in PostgreSQL, but I'm not sure this is worth it.

**Review by renchap — APPROVED** [review]

**Pull request merged**